icu4c/source/test/fuzzer/Makefile.in

```diff
@@ -0,0 +1,99 @@
+# © 2019 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+#******************************************************************************
+## Makefile.in for ICU - test/fuzzer
+
+## Source directory information
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+
+top_builddir = ../..
+
+## All the flags and other definitions are included here.
+include $(top_builddir)/icudefs.mk
+
+## Build directory information
+subdir = test/fuzzer
+
+## Extra files to remove for 'make clean'
+CLEANFILES = *~ $(DEPS) $(TESTXML)
+
+## Target information
+TARGET = fuzzer
+
+BUILDDIR := $(CURR_SRCCODE_FULL_DIR)/../../
+# Simplify the path for Unix
+BUILDDIR := $(BUILDDIR:test/fuzzer/../../=)
+# Simplify the path for Windows
+BUILDDIR := $(BUILDDIR:test\\fuzzer/../../=)
+# Simplify the path for Windows 98
+BUILDDIR := $(BUILDDIR:TEST\\FUZZER/../../=)
+
+CPPFLAGS += -I$(srcdir) -I$(top_srcdir)/common -I$(top_srcdir)/i18n -I$(top_srcdir)/tools/ctestfw -I$(top_srcdir)/io
+DEFS += -D'U_TOPSRCDIR="$(top_srcdir)/"' -D'U_TOPBUILDDIR="$(BUILDDIR)"'
+LIBS = $(LIBCTESTFW) $(LIBICUTOOLUTIL) $(LIBICUIO) $(LIBICUI18N) $(LIBICUUC) $(DEFAULT_LIBS) $(LIB_M)
+
+OBJECTS = fuzzer_driver.o break_iterator_fuzzer.o converter_fuzzer.o locale_fuzzer.o break_iterator_fuzzer.o locale_util.o number_format_fuzzer.o ucasemap_fuzzer.o uloc_canonicalize_fuzzer.o uloc_for_language_tag_fuzzer.o uloc_get_name_fuzzer.o uloc_is_right_to_left_fuzzer.o uloc_open_keywords_fuzzer.o unicode_string_codepage_create_fuzzer.o uregex_open_fuzzer.o
+
+FUZZER_TARGETS = break_iterator_fuzzer converter_fuzzer locale_fuzzer number_format_fuzzer ucasemap_fuzzer uloc_canonicalize_fuzzer uloc_for_language_tag_fuzzer uloc_get_name_fuzzer uloc_is_right_to_left_fuzzer uloc_open_keywords_fuzzer unicode_string_codepage_create_fuzzer uregex_open_fuzzer
+
+DEPS = $(OBJECTS:.o=.d)
+
+-include Makefile.local
+
+## List of phony targets
+.PHONY : all all-local install install-local clean clean-local	\
+distclean distclean-local dist dist-local check check-local xcheck	\
+check-exhaustive check-exhaustive-local fuzzer
```

Comment:
what is the `fuzzer` phony target for?

Reply:
The fuzzer target has the list of real fuzzer executables as precondition, i.e. kicks off the generation of the fuzzer executables. The fuzzer Makfile is different from other test/*/Makefiles insofar that i generates multiple test executables, not just e.g. 'intltest'.

The fuzzer target was declared .PHONY since it does not correspond to a real file. It may also improve performance, no implicit rule search, albeit I am not fully sure this applies here.

Reply:
it probably could be named `all-fuzzers` or something to make it clear what it makes.

Reply:
Done.